### ISSUE TITLE
non-hashable static_argnums are not supported in Jax 0.2.6

### ISSUE THREAD

**Issue by jeremiecoullon** [rlouf/mcx#58]
Jax 0.2.6 stops supporting non-hashable static arguments for jitted functions. See [changelog](https://jax.readthedocs.io/en/latest/CHANGELOG.html#jax-0-2-6-nov-18-2020). That change log points to an explanation [here](https://github.com/google/jax/commit/cb48f42). 

This means that arrays cannot be set as static arguments in jitted functions. An example of a functions that does this at the moment: in `inference.warmup.num_steps_adaptation.py`, the function `run()` defined in the `longest_batch_before_turn()` function sets arrays as static.

This means that cloning mcx and running it with an up-to-date version of jax doesn't work.

**Comment by rlouf** [comment]
Good catch! There's another incompatibility with newer versions of JAX (#36), so I pinned the version temporarily. That's the price to pay when you use a young library :)

Good news is these are easily fixable and won't impact performance 😅
